### ISSUE TITLE
Using Folders on the LCD Screen on the SD Card

### ISSUE THREAD

**Issue by Maximizer18** [bambulab/BambuStudio#1003]
Hi, I'd like to request a feature added where you can put things in folders on the SD card, and see the folders on the LCD screen. I like to stay organized, and that way I know which prints are for which nozzle, material type, multi color, and so on, without naming each and every file what it is. I can just put it in that folder that has only PETG prints, or only PLA prints, and so on.

**Comment by Lesani** [comment]
yes please, I also keep many files on the SD for quick access directly at the printer, and on my prusa machine I can store them by my folder structure. I do store them in a structure on the bambu already, in the hopes of this actually being implemented at some point

**Comment by RobinWhitfield** [comment]
Definitely add me to the list of people desiring folder support.

**Comment by vbkredeemer** [comment]
Here too. Would also be great to have access to that folders to manually add gcode-files. More like klipper. But I see this maybe is the "media"-stuff for X1 and X1C and only for that printers?! Just to have folders, you can use from bambuslicer when sending a file will be a big improvement.

**Comment by FreeManFreeThinker** [comment]
I would love to be able to do the following:

- View/open folders and files in bambu studio that are on the sd card while it's in the printer. Being able to open 3mf files on the sd card, view the project natively in Bambu studio, make changes, and re-save with a different name would be insanely awesome.

- Modify and delete folders/files on the sd card while it's in the printer.

- More control over the 3mf file preview image. Simply add the ability for us to adjust the angle at which the image is taken. Sometimes the thumbnails for the files are taken at an angle that doesn't show all the print/objects on the plate.

**Comment by CCS86** [comment]
Another vote for folder support!

Also, being able to view and modify internal storage and SD storage via BS would be very helpful. the touchscreen on the printer is quite laggy.

**Comment by johnmbaughman** [comment]
+1 for this!

**Comment by sodachen22** [comment]
Thank you for the feedback. An internal requirement id (BBL001-5079) is created, this feature will be evaluated and planned.

**Comment by Maximizer18** [comment]
I see this may be implemented, which is awesome! I have purchased a P1P since I first made this request, and would love this same kind of feature to be able to be used on the P1P as well. Obviously no touch screen, but being able to navigate folders on the screen we have on there would be an awesome feature! Thank you!

**Comment by Benderrodrigu3z** [comment]
> I see this may be implemented, which is awesome! I have purchased a P1P since I first made this request, and would love this same kind of feature to be able to be used on the P1P as well. Obviously no touch screen, but being able to navigate folders on the screen we have on there would be an awesome feature! Thank you!

it has yet to be added

**Comment by Maximizer18** [comment]
> > I see this may be implemented, which is awesome! I have purchased a P1P since I first made this request, and would love this same kind of feature to be able to be used on the P1P as well. Obviously no touch screen, but being able to navigate folders on the screen we have on there would be an awesome feature! Thank you!
> 
> it has yet to be added

I know, but they said it's being "evaluated and planned"

**Comment by homewerken** [comment]
would like to push it again. ore is it still possible?

**Comment by lucian151** [comment]
We cannot express enough how important this feature is for us. We have over 100 files we need to browse through in random order each time we need to kick off a print. This is multiplied across the 6 machines we have.

**Comment by mmuellerphoto** [comment]
It's really annoying to scroll through all the files ... folders please!

**Comment by realmadridsi** [comment]
Can't stress enough how important this is for people like me who use their printers in a print farm. People running print farms have too many products to have them all listed in random order.

**Comment by claytonhoxworth** [comment]
Yes! I am waiting for this as well. My experience running my X1C's would go so much smoother if I could store print files into folders and see the folders on the machine. I have over 170 files that I can sort through and find with no issue on other machine brands with folder support, but it's a pain on my 7 X1C's! Please Bambu add this!

**Comment by iliya-yordanov** [comment]
+1 for folders.

**Comment by napobonaz** [comment]
Another +1 for folders (multiple X1 printers). 
Any status update on this feature? Would be really really helpful.

**Comment by KytorIndustries** [comment]
Any update on this? @sodachen22

**Comment by koder-guy** [comment]
Looking at moving from clipper/mainsail were we had folders and it was great. Really disappointed to find no folder support on Bambu printers?? How is this not possible??. I can't work without organising all my files into folders so without this I will not be purchasing any more Bambu printers. It makes using the printer unbearable.

**Comment by firkateyn82** [comment]
+1 for this function. I can not understand the point of not having folder support. What do they expect having 100s of files in the root?

**Comment by fabiomaticus** [comment]
News for this request? I

**Comment by CCS86** [comment]
Please please please!

**Comment by mackbon401** [comment]
+1 for this! I run a university makerspace and we have 15 XC-1 printers on their way. There are hundreds of students that come through the lab to print things. I'm really hoping this gets resolved by August when we officially open our new space, otherwise it is going to be hectic.

Even our crappy Lulzbot printers have folders!

**Comment by Sissou7** [comment]
+1 please add this support

**Comment by RyanEwen** [comment]
There are quite a few issues / feature requests for this. It's really needed

**Comment by gary-wlg** [comment]
Has there been any movement on this feature? I, as many others have said find it very frustrating to have to move through hundreds of files on each printer to start a job.  Looks like this was opened almost 2 years ago and was being evaluated.

**Comment by CCS86** [comment]
@sodachen22 Can you give us an update please?

**Comment by jryan15** [comment]
+1 sad this is not available on the flagship 3d printer

**Comment by borsch** [comment]
+1

**Comment by oO-Markus-Oo** [comment]
+1 for me. Now where Bambu plans to deactivate MQTT support this becomes even more important. A folder structure would allow a easy start of prints via Handy and / or directly from the printers without the need to slice it everytime.

Sure you can do this also without folders - issue is that you won’t find the gcode you like to print 🤪

**Comment by CCS86** [comment]
Unfortunately, even if they add this feature, which I really doubt at this point, when you start prints off the SD card, you give up all the notifications on Bambu Handy. So, if the printer has some sort of issue, it will not notify you. 

This is such a ridiculous gap in functionality. I have pushed their support to fix this and they won't. They act as though it's not possible for the printer itself to tell "the cloud" that a print has started, and if "the cloud" didn't initiate the print, notifications cannot be sent.

**Comment by untotren** [comment]
HOW IS THIS STILL NOT IMPLEMENTED?!?! THIS IS COMPLETE INEPTITUDE.

**Comment by untotren** [comment]
> Unfortunately, even if they add this feature, which I really doubt at this point, when you start prints off the SD card, you give up all the notifications on Bambu Handy. So, if the printer has some sort of issue, it will not notify you.
> 
> This is such a ridiculous gap in functionality. I have pushed their support to fix this and they won't. They act as though it's not possible for the printer itself to tell "the cloud" that a print has started, and if "the cloud" didn't initiate the print, notifications cannot be sent.
> 
> I have honestly given up hope that they still want to improve these things. Their dev team on here used to be responsive, and they have just checked out and started ignoring everything. Even things they promised to fix or look into. No replies in many months. This is a really bad choice when the market is becoming more competitive from a hardware perspective.
> 
> I will be getting a different brand for my upcoming printer because of this.

Let me know what you switch to please, I'd love to buy something from a more responsive company.
